File: src/main/scala/it/carloni/luca/aurora/option/ScoptOption.scala
package it.carloni.luca.aurora.option

object ScoptOption extends Enumeration {

  protected case class Val(short: Char, long: String, text: String) extends super.Val

  import scala.language.implicitConversions
  implicit def valueToVal(x: Value): Val = x.asInstanceOf[Val]

  val applicationBranchOption: Val = Val('b', "branch", "Application branch to run")
  val propertiesOption: Val = Val('p', "properties", "Name of .properties file")
  val sourceOption: Val = Val('s', "source", "Source name (BANCLL) for which ingestion must be triggered")
  val businessDateOption: Val = Val('d', "date", "Working business date")
  val mappingSpecificationFlag: Val = Val('m', "mapping_specification", "Flag for overwriting mapping specification table")
  val lookUpSpecificationFlag: Val = Val('l', "look_up", "Flag for overwriting look up table")
  val completeOverwriteFlag: Val = Val('o', "overwrite", "Flag for specifing behavior when reloading tables. " +
    "True: drop and reload. False: truncate and reload")
}

File: src/main/java/it/carloni/luca/aurora/option/ScoptOption.java
package it.carloni.luca.aurora.option;

public enum ScoptOption {

    APPLICATION_BRANCH('b', "branch", "Application branch to run"),
    PROPERTIES_OPTION('p', "properties", "Name of .properties file"),
    SOURCE_OPTION('s', "source", "Source name (BANCLL) for which ingestion must be triggered"),
    BUSINESS_DATE_OPTION('d', "date", "Working business date"),
    MAPPING_SPECIFICATION_FLAG('m', "mapping_specification", "Flag for overwriting mapping specification table"),
    LOOKUP_SPECIFICATION_FLAG('l', "look_up", "Flag for overwriting look up table"),
    COMPLETE_OVERWRITE_FLAG('o', "overwrite", "Flag for specifing behavior when reloading tables. " +
            "True: drop and reload. False: truncate and reload"),
    VERSION_NUMBER_OPTION('v', "version", "Specification version number to be referred to");

    private final char shortOption;
    private final String longOption;
    private final String description;

    public char getShortOption() { return shortOption; }
    public String getLongOption() { return longOption; }
    public String getDescription() { return description; }

    ScoptOption(char shortOption, String longOption, String description) {

        this.shortOption = shortOption;
        this.longOption = longOption;
        this.description = description;
    }
}
